main/src/ca/uwaterloo/flix/language/phase/Redundancy.scala

```diff
@@ -50,9 +50,17 @@ object Redundancy extends Phase[TypedAst.Root, TypedAst.Root] {
       return root.toSuccess
     }
 
+    val instanceDefs = for {
```

Comment:
I would just ParOps.parAgg over the instances. Then over the defs. And then merge them at the end.

I know this means that there is a little bit less parallelism than theoretically possible, but I don't think this will be measurable, and I would rather have that the code structure is easier to see. Moreover, I am not sure if using `++` does not impose its own costs. (Maybe not if they are iterables at this point...)